vars/runTest.groovy

```diff
@@ -99,6 +99,10 @@ def call(Map config = [:]) {
                           config['failure_artifacts'] + '"'
     }
 
+    if(config['with_valgrind']) {
+        env.WITH_VALGRIND = config['with_valgrind']
```

Comment:
@JohnMalmberg @brianjmurrell on my latest review and jenkins job I'm getting the same old error of having both `run_test.sh` and `Unit Test with memcheck` stages run with memcheck 

This is run_test.sh complete log https://build.hpdd.intel.com/blue/rest/organizations/jenkins/pipelines/daos-stack/pipelines/daos/branches/PR-3023/runs/67/nodes/407/steps/487/log/?start=0
and this is UT with memcheck https://build.hpdd.intel.com/blue/rest/organizations/jenkins/pipelines/daos-stack/pipelines/daos/branches/PR-3023/runs/67/nodes/408/steps/486/log/?start=0

This was solved before by setting up the environment variable here in `runTest.groovy`, but it seems to not working anymore, do you know how to fix this?

Reply:
I don't think you can persistently set an environment variable this way, and if you can, it will probably introduce a bug.

It is a matter of variable scope, which is one of our common problems in trying to debug flaky behavior in groovy scripts and Jenkinsfiles.

In a groovy library if you do not declare a variable as "def" or a type, that variable becomes global to the entire Jenkins job.  Which means that if you have two unit test stages running in parallel and are missing that declaration somewhere, sometimes it will work, and sometimes it will not.

I am not sure what scope the "env" variable is in.

Usually I debug groovy scripts by adding println statements in them to dump variables and let me know why a conditional path is taken.